atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/InstrumentedCassandraClient.java

```diff
@@ -0,0 +1,170 @@
+
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace(
+                "client.multiget_slice(table {}, number of keys {}, number of columns {}, consistency {})",
+                LoggingArgs.safeInternalTableNameOrPlaceholder(internalTableReference),
+                numberOfKeys, numberOfColumns, consistency_level)) {
+            return registerDuration(
+                    () -> delegate.multiget_slice(keys, column_parent, predicate, consistency_level),
+                    MULTIGET_SLICE_TIMER,
+                    logger -> logger.warn("client.multiget_slice({}, {}, {}, {})",
+                            LoggingArgs.safeInternalTableNameOrPlaceholder(internalTableReference),
+                            SafeArg.of("number of keys", numberOfKeys),
+                            SafeArg.of("number of columns", numberOfColumns),
+                            SafeArg.of("consistency", consistency_level.toString())));
+        }
+    }
+
+    @Override
+    public List<KeySlice> get_range_slices(ColumnParent column_parent, SlicePredicate predicate, KeyRange range,
+            ConsistencyLevel consistency_level)
+            throws InvalidRequestException, UnavailableException, TimedOutException, TException {
+        //noinspection unused - try-with-resources closes trace
+        // TODO(ssouza): also log the row key names when they can be marked as safe for logging.
+        String internalTableRef = column_parent.column_family;
+        int numberOfKeys = predicate.slice_range.count;
+        int numberOfColumns = range.count;
+
+        try (CloseableTrace trace = startLocalTrace(
+                "client.get_range_slices(table {}, number of keys {}, number of columns {}, consistency {})",
+                LoggingArgs.safeInternalTableNameOrPlaceholder(internalTableRef),
+                numberOfKeys, numberOfColumns, consistency_level)) {
+            return registerDuration(
+                    () -> delegate.get_range_slices(column_parent, predicate, range, consistency_level),
+                    GET_RANGE_SLICE_TIMER,
+                    logger -> logger.warn("client.get_range_slices({}, {}, {}, {})",
+                            LoggingArgs.safeInternalTableNameOrPlaceholder(internalTableRef),
+                            SafeArg.of("number of keys", numberOfKeys),
+                            SafeArg.of("number of columns", numberOfColumns),
+                            SafeArg.of("consistency", consistency_level.toString())));
+        }
+    }
+
+    private <T> T registerDuration(CallableTException<T> callable, Timer timer, Consumer<Logger> loggerFunction)
+            throws TException {
+
+        long startTime = System.nanoTime();
+        T ret = callable.call();
+        long endTime = System.nanoTime();
+        long duration = endTime - startTime;
+        timer.update(duration, TimeUnit.NANOSECONDS);
```

Comment:
might want to try-with-resources using `timer.time()`, or at least make sure we're timing potential exception cases as those are important to capture if we're timing out.